SPECIFICATION.md

```diff
@@ -63,61 +123,223 @@ The minimum difficulty that the pool supports. This will also be the default tha
+The pool should reject requests made with outdated `authentication_keys`. This key can be changed using `PUT /farmer`,
+which is signed with the owner key.
 
-Failed response:
+#### payload.payout_instructions
+This is the instructions for how the farmer wants to get paid. By default this will be an XCH address, but it can
+be set to any string with a size of less than 1024 characters, so it can represent another blockchain or payment
+system identifier.
+
+#### payload.suggested_difficulty
+A request from the farmer to update the difficulty. Can be ignored or respected by the pool. However, this should only
+be respected if the authentication public key is the most recent one seen for this farmer.
+
+#### signature
+This is a BLS signature of the hashed serialization of the payload:
+
+```
+sha256(PostFarmerPayload)
+```
+
+signed by the private key of the `owner_public_key` using the Augmented Scheme in the BLS IETF spec.
+
+See the [streamable](#signature-validation) class `PostFarmerPayload` in the
+[pool protocol](https://github.com/Chia-Network/chia-blockchain/blob/main/chia/protocols/pool_protocol.py)
+and [Farmer authentication](#farmer-authentication) for the specification of `authentication_token`.
+
+## PUT /farmer
+Allows farmers to update their information on the pool.
+
+Request:
 ```json
-{"error_code": 4, "error_message": "Invalid proof of space"}
+{
+    "payload": {
+        "launcher_id": "0xae4ef3b9bfe68949691281a015a9c16630fc8f66d48c19ca548fb80768791afa",
+        "authentication_token": 27062279,
+        "authentication_public_key": "0x970e181ae45435ae696508a78012dc80548c334cf29676ea6ade7049eb9d2b9579cc30cb44c3fd68d35a250cfbc69e29",
+        "payout_instructions": "0xc2b08e41d766da4116e388357ed957d04ad754623a915f3fd65188a8746cf3e8",
+        "suggested_difficulty": 10
+    },
+    "signature": "0xa078dc1462bbcdec7cd651c5c3d7584ac6c6a142e049c7790f3b0ee8768ed6326e3a639f949b2293469be561adfa1c57130f64334994f53c1bd12e59579e27127fbabadc5e8793a2ef194a5a22ac832e92dcb6ad9a0d33bd264726f6e8df6aad"
+}
+```
+
+For a description of the request body entries see the corresponding keys in [POST /farmer](#post-farmer). The values
+provided with the key/value pairs are used to update the existing values. All entries, except `launcher_id`, are
+optional but there must be at least one of them. 
+
+See the [streamable](#signature-validation) class `PutFarmerPayload` in the
+[pool protocol](https://github.com/Chia-Network/chia-blockchain/blob/main/chia/protocols/pool_protocol.py) for details
+and [Farmer authentication](#farmer-authentication) for the specification of
+`authentication_token`.
+
+Successful response:
+```json
+{
+  "authentication_public_key": true,
+  "payout_instructions": true,
+  "suggested_difficulty": true
+}
+```
+
+The successful response must always contain one key/value pair for each entry provided in the request body. The value
```

Comment:
`The successful response` -> `A successful response`

I think there is one case that English and German differ!